File: 백준/Silver/9012. 괄호/괄호.py
import sys
input = sys.stdin.readline

T = int(input())

for _ in range(T):
    ps = input().strip()
    count = 0
    
    for char in ps:
        if char == '(':
            count += 1
        elif char == ')':
            count -= 1
        
        if count < 0:
            break

    if count == 0:
        print("YES")
    else:
        print("NO")

File: 백준/Silver/9012. 괄호/괄호.js
const input = require('fs').readFileSync(0).toString().trim().split('\n');
const N = input.shift();

for (let ps of input) {
    let count = 0;
    
    for (let char of ps) {
        if (char === '(') {
            count += 1;
        } else if (char === ')') {
            count -= 1;
        }
        
        if (count < 0) {
            break;
        }
    }
    
    console.log(count === 0 ? 'YES' : 'NO');
}